spec/controllers/dashboard_controller_spec.rb

```diff
@@ -119,4 +119,29 @@
       end
     end
   end
+
+  describe 'Check Primary Account' do
+    context 'login as subsidiary account' do
+      it 'renders unauthenticated subsidiary haml' do
+        when_current_user_is :student
+        admin = create :user, :admin
+        request.env.merge!({ 'uid' => session[:user_id], 
```

Comment:
It will, but you have to drop the curly braces too.  This is because curly braces are also the way ruby syntax specifies a block argument to a method, but what you have isn't a valid block.

```ruby
request.env.merge! 'uid' => 'session[:user_id],
                   'UMAPrimaryAccount' => admin.id
```

Should be valid, I think.